packages/vscode/src/index.ts

```diff
@@ -91,6 +104,50 @@ export async function activate(context: ExtensionContext) {
 	context.subscriptions.push(
 		commands.registerCommand('astro.restartLanguageServer', async () => {
 			await restartClient(true);
+		}),
+		commands.registerCommand('astro.openDevServerPanel', async () => {
+			// Let's first try to find if we have a dev server currently running already
+			let url = await getDevServerUrl();
+
+			// If not, let's run one
+			if (!url) {
+				const terminal = window.createTerminal({ name: 'astro:dev' });
+				terminal.show();
+				terminal.sendText('node ./node_modules/astro/astro.js dev');
+
+				// Wait for the dev server to start
+				await sleep(2500);
+
+				url = await getDevServerUrl();
+
+				if (!url) {
+					window.showErrorMessage(
+						'Could not find a local server. This might be because the dev server took too long to start'
+					);
+					return;
+				}
+			}
+
+			commands.executeCommand('simpleBrowser.api.open', url, {
+				viewColumn: ViewColumn.Beside,
+			});
+
+			/**
+			 * Return a currently running dev server's URL
+			 */
+			async function getDevServerUrl() {
```

Comment:
Would be super cool if the page in the browser synced with the current page you're editing... too wild?

Reply:
That'd be super cool! We have all the information needed to do that, I think 🤔

Reply:
^ Would this work similarly to how the [astro-repl](https://astro.build/play) does?